### PR TITLE
Aqaint 182

### PR DESCRIPTION
**Pull request by vkrasnobelmov**

### CODE REVIEW

File: BookingSpecBindings/Bindings/SignInPopUpBindings.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		}
+		[Then(@"I click on X button")]
+		public void ThenIClickOnXButton()
+		{
+			signPage.CloseSignInPopUp();
+		}
+		[Then(@"I see message error with ""(.*)""")]
+		public void ThenISeeMessageErrorWith(string message)
+		{
+			Assert.That(signPage.GetErrorText().Contains(message), Is.True);
```
Comment (IgorRukhlenko) [inline BookingSpecBindings/Bindings/SignInPopUpBindings.cs]: Contains() is not enough here. It's better to compare an entire message to catch possible issues with text.

File: BookingSpecTests/LoginFeature.feature [hunk 2]
```diff
@@ -0,0 +1,125 @@
+﻿Feature: Login
```
Comment (IgorRukhlenko) [inline BookingSpecTests/LoginFeature.feature]: Please fix usage of When and Then in this feature file.

File: BookingSpecTests/LoginFeature.feature [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+Scenario: Cancel recover password and Sign in
+	When I open browser
+	And I navigate to url "https://booking.com"
+	When I click Sign In button on Ribbon menu
+	When I click Forgot Your Password button
+	When I set following parameters on Forgot Your Password Pop Up dialog
+		| Field    | Value                      |
+		| Email    | testmirantistest@gmail.com |
+	Then I click button Cancel on Forgot Your Password PopUp
```
Comment (IgorRukhlenko) [inline BookingSpecTests/LoginFeature.feature]: Need to check that Cancel button will lead to the expected system behavior.

File: BookingSpecTests/LoginFeature.feature [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	And I wait while page popup is loading
+	Then I click button Back To Sign In on Forgot Your Password PopUp
+	When I set following parameters on Sign In Pop Up dialog
+		| Field    | Value                      |
+		| Email    | testmirantistest@gmail.com |
+		| Password | 1234qweR                   |
+	And I click Sign In button on PopUp
+	Then I see that I am Signed In 
+
+Scenario: Check that previous values in PopUp fields are saved
```
Comment (IgorRukhlenko) [inline BookingSpecTests/LoginFeature.feature]: There is no actual check that previous values are saved in this scenario.

File: BookingSpecTests/LoginFeature.feature [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	When I set following parameters on Sign In Pop Up dialog
+		| Field    | Value                      |
+		| Email    | testmirantistest@gmail.com |
+		| Password | 1234qweR                   |
+	Then I click on X button
+	And  I click Sign In button on Ribbon menu
+	When I click Sign In button on PopUp
+	Then I see that I am Signed In 
+
+Scenario: Check that previous values in PopUp fields are not saved
```
Comment (IgorRukhlenko) [inline BookingSpecTests/LoginFeature.feature]: There is no actual check that previous values are not saved in this scenario.

**Review by IgorRukhlenko — CHANGES_REQUESTED** [review]

**Review by IgorRukhlenko — APPROVED** [review]

File: BookingSpecBindings/Bindings/ForgotYourPasswordPopUpBindings.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+				}
+			}
+		}
+		[When(@"I wait while page popup is loading")]
+		public void WhenIWaitWhilePagePopupIsLoading()
+		{
+			forgotPage.waitLoading();
+		}
+
+		[Then(@"I wait while page popup is loading")]
```
Comment (etolmachev) [inline BookingSpecBindings/Bindings/ForgotYourPasswordPopUpBindings.cs]: Dublicated methods

File: BookingSpecBindings/Bindings/SignInPopUpBindings.cs [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		}
+		[Then(@"I see error message ""(.*)"" on Sign In PopUp")]
+		public void ThenISeeErrorMessageOnSignInPopUp(string error)
+		{
+			Assert.AreEqual(signPage.GetErrorText(), error);
+		}
+		[Then(@"I see that credential values are saved in Sign In PopUp fields")]
+		public void ThenISeeThatCredentialValuesAreSavedInSignInPopUpFields()
+		{
+			Assert.AreEqual(signPage.GetEmailText(), "testmirantistest@gmail.com");
```
Comment (etolmachev) [inline BookingSpecBindings/Bindings/SignInPopUpBindings.cs]: Use parameter for email, store email value in FF

**Review by etolmachev — CHANGES_REQUESTED** [review]

**Pull request merged**